projects/project.cpp

```diff
@@ -493,16 +500,23 @@ namespace projects {
          )
       };
 
-      if(
-         // If this cell is planned to remain at the current refinement level, but is outside the allowed refinement region,
-         // attempt to unrefine it instead. (If it is already at the lowest refinement level, DCCRG should not go belly-up.)
-         // Induced refinement still possible just beyond that limit.
-         (xyz[0] < P::refinementMinX) || (xyz[0] > P::refinementMaxX)
-         || (xyz[1] < P::refinementMinY) || (xyz[1] > P::refinementMaxY)
-         || (xyz[2] < P::refinementMinZ) || (xyz[2] > P::refinementMaxZ)) {
-         shouldUnrefine = true;
+      bool shouldNotUnrefine = false;
+      for (uint i = 0; i < P::refineBoxNumber; i++) {
+         if(
+            // If this cell is planned to remain at the current refinement level, but is outside the allowed refinement region,
+            // attempt to unrefine it instead. (If it is already at the lowest refinement level, DCCRG should not go belly-up.)
+            // Induced refinement still possible just beyond that limit.
+            (xyz[0] < P::refinementMinX.at(i)) || (xyz[0] > P::refinementMaxX.at(i))
+            || (xyz[1] < P::refinementMinY.at(i)) || (xyz[1] > P::refinementMaxY.at(i))
+            || (xyz[2] < P::refinementMinZ.at(i)) || (xyz[2] > P::refinementMaxZ.at(i))) {
+            shouldNotUnrefine = shouldNotUnrefine || true;
+         } else {
+            shouldNotUnrefine = shouldNotUnrefine || false;
+         }
       }
 
+      shouldUnrefine = shouldUnrefine || !shouldNotUnrefine;
+
       return shouldUnrefine;
```

Comment:
And this new mechanism says it shouldNotUnrefine if outside any of the defined boxes. Which confuses me. Why shouldn't it coarsen? Especially in your operational case at hand, you'll want things outside of the box to be coarsened, no?

Reply:
Same here, we reviewed this together, I am less confused and the upcoming changes will make things more betterly clearer.